clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/cache/AzureServerGroupCachingAgent.groovy

```diff
@@ -85,7 +85,45 @@ class AzureServerGroupCachingAgent extends AzureCachingAgent {
       it.attributes.processedCount = (it.attributes.processedCount ?: 0) + 1
     }
 
-    result
+    if (result.cacheResults[AZURE_SERVER_GROUPS.ns]) {
+      result
+    } else {
+      // run the cache cleanup routine on an empty server group list only for now
```

Comment:
What if the user has deleted a server group from the portal but there are still other server groups that are getting cached? How will this handle "cleaning up" the deleted server group if it only runs on an empty list?